libtenzir/builtins/aggregation-functions/all.cpp

```diff
@@ -51,7 +56,40 @@ class all_function final : public aggregation_function {
   std::optional<bool> all_ = {};
 };
 
-class plugin : public virtual aggregation_function_plugin {
+class all_instance final : public aggregation_instance {
+public:
+  explicit all_instance(ast::expression expr) : expr_{std::move(expr)} {
+  }
+
+  auto update(const table_slice& input, session ctx) -> void override {
+    auto arg = eval(expr_, input, ctx);
+    if (all_ and all_.value() == false) {
+      return;
+    }
+    auto f = detail::overload{
+      [](const arrow::NullArray&) {},
+      [&](const arrow::BooleanArray& array) {
+        all_ = array.false_count() == 0;
+      },
```

Comment:
If the input is `x`, `y`, `z`, then `all` should return the same as `x and y and z`, where `true and null == null` and `false and null == false`. The evaluator of `and` currently returns "not implemented yet", but we will implement that soon, and this here should directly get the correct behavior :+1:

Reply:
What about `any`? What does `or`ing with `null` return?

Reply:
Yes, `any` uses `or` with `true or null == true` and `false or null == null`.

We are basically implementing this: https://en.wikipedia.org/wiki/Three-valued_logic#Kleene_and_Priest_logics